packages/cascara/src/ui/Table/context/RowProvider.js

```diff
@@ -4,16 +4,24 @@ import { ModuleContext, ModuleProvider } from '../../../modules/context';
 const RowProvider = ({ children, value, ...props }) => {
   const grandparentValues = useContext(ModuleContext);
 
-  // Here we need to not pass the isEditing and setIsEditing values from grandparent which
-  // will override what is happening in the ModuleProvider we pass mergedValues into. This
-  // way the row provider itself can have its own isEditing. This is also a good place for
-  // us to control how these values get set if the overall table is getting edited on
-  // another row.
-  const { isEditing, setIsEditing, ...rest } = grandparentValues;
+  const {
+    idOfRecordInEditMode,
+    uniqueIdAttribute,
+    ...rest
+  } = grandparentValues;
+  const { record } = value;
+  const recordId = record[uniqueIdAttribute];
+
+  // isEditing is based on wether the record ids are the same
+  const isEditing = recordId === idOfRecordInEditMode;
+
```

Comment:
Oh yes. This is great. Now those Modules don't need to know at all what is happening outside of them. Great use case for separation of concerns. Hoist the isEditing logic up a little higher. <3